tests/unit/test_charm.py

```diff
@@ -269,3 +313,249 @@ def test_given_any_config_when_on_config_changed_then_systemd_manager_configurat
         patch_run.assert_any_call(
             "systemctl daemon-reload", shell=True, stdout=-1, encoding="utf-8"
         )
+
+    @patch("subprocess.run")
+    @patch("builtins.open", new_callable=mock_open)
+    def test_given_any_config_and_installed_when_on_config_changed_then_status_is_active(  # noqa: E501
+        self, _, __
+    ):
+        key_values = {}
+        self.harness.charm._stored.installed = True
+
+        self.harness.update_config(key_values=key_values)
+
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus("SW installed."))
+
+    @patch("subprocess.run")
+    @patch("builtins.open", new_callable=mock_open)
+    def test_given_any_config_and_not_installed_when_on_config_changed_then_status_is_active(  # noqa: E501
+        self, _, __
+    ):
+        key_values = {}
+
+        self.harness.update_config(key_values=key_values)
+
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus(""))
+
+    @patch("subprocess.run")
+    @patch("builtins.open", new_callable=mock_open)
+    def test_given_any_config_and_started_is_true_when_on_config_changed_then_srsenb_service_is_restarted(  # noqa: E501
+        self, _, patch_subprocess_run
+    ):
+        key_values = {}
+        self.harness.charm._stored.started = True
```

Comment:
I think this could also be changed to directly check if the workload is started. This could be in another PR however.

Reply:
I agree. Similar to my reply in the comment above.